File: laravel/app/Services/Content/ArticleService.php
<?php

namespace App\Services\Content;

use App\Models\Content\Article;

class ArticleService
{
    public function create(array $data): Article
    {
        throw new \LogicException('Not implemented');
    }

    public function update(Article $article, array $data): Article
    {
        throw new \LogicException('Not implemented');
    }
    public function publish(int $id): void
    {
        $article = Article::findOrFail($id);
        $article->publish();
        $article->save();
    }

    public function archive(int $id): void
    {
        $article = Article::findOrFail($id);
        $article->archive();
        $article->save();
    }

    public function incrementView(int $id): void
    {
        $article = Article::findOrFail($id);
        $article->incrementView();
        $article->save();
    }

    public function readingTimeMinutes(int $id): int
    {
        $article = Article::findOrFail($id);
        $result = $article->readingTimeMinutes();
        $article->save();
        return $result;
    }
}

File: rails/app/services/content/article_service.rb
module Content
  class ArticleService
    def initialize(repository = Article)
      @repository = repository
    end

    def create(attributes)
      raise NotImplementedError, "#{self.class}#create not implemented"
    end

    def update(entity, attributes)
      raise NotImplementedError, "#{self.class}#update not implemented"
    end

    def publish(id)
      instance = Article.find(id)
      instance.publish()
      instance.save!
    end

    def archive(id)
      instance = Article.find(id)
      instance.archive()
      instance.save!
    end

    def increment_view(id)
      instance = Article.find(id)
      instance.increment_view()
      instance.save!
    end

    def reading_time_minutes(id)
      instance = Article.find(id)
      result = instance.reading_time_minutes()
      instance.save!
      result
    end
  end
end
